templates/demo-store/server.ts

```diff
@@ -58,6 +60,41 @@ export default {
         setCartId: cartSetIdDefault(),
       });
 
+      const withCache = createWithCache({
+        cache,
+        waitUntil,
+        request,
+      });
+
+      const catFact = async () => {
+        const data = await withCache<
+          [
+            {
+              fact: string;
+              length: number;
+            },
+            ResponseInit,
+          ]
+        >(
+          ['https://catfact.ninja/fact'],
+          CacheLong(),
+          () => {
+            return fetch('https://catfact.ninja/fact').then(async (res) => {
+              return [
+                (await res.json()) as {fact: string; length: number},
+                {
+                  status: res.status,
+                  statusText: res.statusText,
+                  headers: Array.from(res.headers.entries()),
+                } satisfies ResponseInit,
+              ];
+            });
+          },
+          'Cat fact',
+        );
+        return data[0];
+      };
```

Comment:
Example usage with `createWithCache`. (Debatable topic)

Without making a breaking change to `createWithCache`, this is what it is going to look like if devs want to get the server network timing tool working with all the information for 3p queries.

It's a bit messy but we don't have a type structure on what `actionFn` should return.

Reply:
Note that `storefront.query` also returns a structure like `[any, ResponseInit]`